vic/drivers/classic/src/get_force_type.c

```diff
@@ -147,3 +147,50 @@ get_force_type(char *cmdstr,
 
     (*field)++;
 }
+
+
+/******************************************************************************
+ * @brief    This routine determines the counts the number of forcing variables
+             in each forcing file specified in the global parameter file.
+ *****************************************************************************/
+size_t
+count_force_vars(FILE *gp)
+{
+    size_t        nvars;
+    unsigned long start_position;
+    char          cmdstr[MAXSTRING];
+    char          optstr[MAXSTRING];
+    // Figure out where we are in the input file
+    fflush(gp);
+    start_position = ftell(gp);
+
+    // read the first line
+    fgets(cmdstr, MAXSTRING, gp);
+
+    // initalize nvars
+    nvars = 0;
+
+    // Loop through the lines
+    while (!feof(gp)) {
+        if (cmdstr[0] != '#' && cmdstr[0] != '\n' && cmdstr[0] != '\0') {
+            // line is not blank or a comment
+            sscanf(cmdstr, "%s", optstr);
+
+            // if the line starts with FORCE_TYPE
+            if (strcasecmp("FORCE_TYPE", optstr) == 0) {
+                nvars++;
+            }
+            // else if we arive at another forcing file break out of loop
+            else if (strcasecmp("FORCING1", optstr) == 0 ||
```

Comment:
I think this is a fine idea.  We already have the infrastructure to support N forcing file streams.  No reason not to generalize it.  I've opened a new issue for it in #383.